### PR TITLE
[build] Add "make check" and "make quickcheck" build targets

### PR DESCRIPTION
**Pull request by grgustaf**
The check target calls trlite with a new -l flag that makes it run
on the current ZJS_BASE source tree w/o doing a git clone for a clean
tree.

The quickcheck calls trlite on just the Linux tests (VM3).

Signed-off-by: Geoff Gustafson <geoff@linux.intel.com>

### CODE REVIEW

File: scripts/trlite [hunk 1]
```diff
@@ -6,8 +6,10 @@
 # trlite - a local version of the tests we run in Travis
 #   trlite [-j] [-nN] [-v] [vmname] [startnum] [endnum]
 #
-#   -nN appends N to .trlite directory name
+#   (for now, options much appear in specified order)
```
Comment (pfalcon) [inline scripts/trlite]: "must appear" I guess

File: scripts/trlite [hunk 2]
```diff
@@ -86,6 +101,11 @@ if [[ $1 == -j* ]]; then
 fi
 
 if [[ $1 == -n* ]]; then
+    if [ "$RUNLOCAL" == "y" ]; then
+        echo Error: -j mode cannot be used with -l
```
Comment (pfalcon) [inline scripts/trlite]: "-n mode cannot" I guess?

Comment (grgustaf) [inline scripts/trlite]: Thanks, right.

**Comment by pfalcon** [comment]
Ok, ran to completion for me too, "quickcheck" is definitely useful, and I should become even more useful with https://github.com/01org/zephyr.js/pull/895.

"check" is somwhat longish (12min) for me, and I need to dig what exactly it does (besides just building all (or many) samples), but good to know it's there.

So, I'd suggest to augment this patch with adding both "quickcheck" and "check" to "make help" - at the very least, and I'd suggest carving out couple of lines in README to mention them too. Otherwise: +1, thanks!

File: README.md [hunk 3]
```diff
@@ -236,6 +236,23 @@ least two plus-ones (+1) , "Look Good To Me (LGTM)" or other positive signals
 for the project members. Once you have gained the required signals the project
 maintainers will merge the PR.
 
+### Travis and local sanity checking
+
+We run a series of tests on each pull request and merged commit using Travis.
+This relies on a script in the repo called `trlite`. One easy way to run these
```
Comment (pfalcon) [inline README.md]: Perhaps spell as `scripts/trlite`.

**Comment by pfalcon** [comment]
+1

**Comment by jimmy-huang** [comment]
+1

**Pull request merged**